instrumentation/kafka/kafka-clients/kafka-clients-common/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/KafkaMetricRegistry.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import static io.opentelemetry.instrumentation.kafkaclients.InstrumentDescriptor.INSTRUMENT_TYPE_DOUBLE_OBSERVABLE_COUNTER;
+import static io.opentelemetry.instrumentation.kafkaclients.InstrumentDescriptor.INSTRUMENT_TYPE_DOUBLE_OBSERVABLE_GAUGE;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.ObservableDoubleMeasurement;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.Measurable;
+
+/** A registry mapping kafka metrics to corresponding OpenTelemetry metric definitions. */
+class KafkaMetricRegistry {
+
+  private static final Set<String> groups = new HashSet<>(Arrays.asList("consumer", "producer"));
+  private static final Map<Class<?>, String> measureableToInstrumentType = new HashMap<>();
+  private static final Map<String, String> descriptionCache = new ConcurrentHashMap<>();
+
```

Comment:
can these all be instance fields (non-static) and store KafkaMetricRegistry instance in the MetricsReporter instance?

Reply:
Yes, although KafkaMetricRegistry would have to be held in a static variable in the MetricReporter so there won't be much difference.

Reply:
it looked like it could be held in an instance variable in MetricReporter since only accessed from instance method there?

Reply:
I suppose it could, but there's no advantage because `KafkaMetricRegistry` is nearly stateless. The only state it has is `descriptionsCache`, which needs to be static. 

Similarly, the `OpenTelemetryKafkaMetrics` implementation of `MetricReporter` needs to hold its registry of observables in a static variable (`List<RegisteredObservable> registeredObservables`) to ensure that there is only one callback registered for each unique metric identity even when `OpenTelemetryKafkaMetrics` is registered to multiple producers / consumers.